src/sagemaker/model.py

```diff
@@ -477,8 +477,10 @@ def deploy(
 
         compiled_model_suffix = "-".join(instance_type.split(".")[:-1])
         if self._is_compiled_model:
-            name_prefix = self.name or utils.name_from_image(self.image)
-            self.name = "{}{}".format(name_prefix, compiled_model_suffix)
+            name_prefix = self.name or utils.name_from_image(
+                self.image, max_length=(62 - len(compiled_model_suffix))
```

Comment:
Should this be 63 - len(compiled_model_suffix) ?

Reply:
I added a dash to join the prefix and the suffix, so that's why I took one off from the 63 limit